src/amo/pages/UserProfileEdit/index.js

```diff
@@ -198,8 +200,24 @@ export class UserProfileEditBase extends React.Component<Props, State> {
       });
     }
 
-    if (wasUpdating && !isUpdating && !errorHandler.hasError()) {
+    const redirectToUserProfile = () => {
       history.push(`/${lang}/${clientApp}/user/${newUserId}/`);
+    };
+
+    if (wasUpdating && !isUpdating && !errorHandler.hasError()) {
+      let toPath = location.query.to;
+      if (toPath && typeof toPath === 'string' && !toPath.startsWith('//')) {
+        if (!toPath.startsWith('/')) {
+          toPath = `/${toPath}`;
+        }
+        try {
+          history.push(toPath);
+          return;
+        } catch (error) {
+          log.warn(`Error redirecting to location: ${toPath}: ${error}`);
+        }
+      }
+      redirectToUserProfile();
```

Comment:
This code would be easier to read if it was just `history.push(...)` instead of a function call.